server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSubmissionSetMethod.scala

```diff
@@ -0,0 +1,256 @@
+      case e: ForbiddenFromException => SetError(EmailSubmissionSetMethod.forbiddenFrom,
+        SetErrorDescription(s"Attempt to send a mail whose envelope From not allowed for connected user: ${e.from}"),
+        Some(Properties("envelope.mailFrom")))
+      case _: MessageNotFoundException => SetError(SetError.invalidArgumentValue,
+        SetErrorDescription("The email to be sent cannot be found"),
+        Some(Properties("emailId")))
+      case e: Exception =>
+        e.printStackTrace()
+        SetError.serverFail(SetErrorDescription(exception.getMessage))
+    }
+  }
+  case class CreationResults(created: Seq[CreationResult]) {
+    def retrieveCreated: Map[EmailSubmissionCreationId, EmailSubmissionCreationResponse] = created
+      .flatMap {
+        case success: CreationSuccess => Some(success.emailSubmissionCreationId, success.emailSubmissionCreationResponse)
+        case _ => None
+      }
+      .toMap
+      .map(creation => (creation._1, creation._2))
+
+
+    def retrieveErrors: Map[EmailSubmissionCreationId, SetError] = created
+      .flatMap {
+        case failure: CreationFailure => Some(failure.emailSubmissionCreationId, failure.asSetError)
+        case _ => None
+      }
+      .toMap
+  }
+
+  def init(): Unit = {
+    queue = mailQueueFactory.createQueue(SPOOL)
+  }
+
+  @PreDestroy def dispose(): Unit = {
+    Try(queue.close())
+      .recover(e => LOGGER.debug("error closing queue", e))
+  }
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: EmailSubmissionSetRequest): SMono[InvocationWithContext] = {
+    for {
+      createdResults <- create(request, mailboxSession, invocation.processingContext)
+    } yield InvocationWithContext(
+      invocation = Invocation(
+        methodName = invocation.invocation.methodName,
+        arguments = Arguments(serializer.serializeEmailSubmissionSetResponse(EmailSubmissionSetResponse(
+            accountId = request.accountId,
+            newState = State.INSTANCE,
+            created = Some(createdResults._1.retrieveCreated).filter(_.nonEmpty),
+            notCreated = Some(createdResults._1.retrieveErrors).filter(_.nonEmpty)))
+          .as[JsObject]),
+        methodCallId = invocation.invocation.methodCallId),
+      processingContext = createdResults._2)
+  }
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): SMono[EmailSubmissionSetRequest] =
+    asEmailSubmissionSetRequest(invocation.arguments)
+
+  private def asEmailSubmissionSetRequest(arguments: Arguments): SMono[EmailSubmissionSetRequest] =
+    serializer.deserializeEmailSubmissionSetRequest(arguments.value) match {
+      case JsSuccess(emailSubmissionSetRequest, _) => SMono.just(emailSubmissionSetRequest)
+      case errors: JsError => SMono.raiseError(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
+    }
```

Comment:
the private method only used once? May be the logic can be written in getRequest() directly?